### PR TITLE
use gh-issues to update conferences.json

### PR DESCRIPTION
**Pull request by kjaymiller**
- update issue template to include start and end dates for conference
- update conferences to use `gh-issues` instead of manual parsing
- update conferences template to use `grid` and new json data

- add conferences.json

### CODE REVIEW

**Comment by kjaymiller** [comment]
TODO:
  - [x] Update Tests

**Comment by kjaymiller** [comment]
@dragid10 

You were the last to work on this. would you mind puttins some eyes on this change.

File: _conferences/__main__.py
Comment (dragid10) [inline _conferences/__main__.py:7]: Me looking up the `gh_issues` library
image
skeptical-futurama

File: _conferences/__main__.py
Comment (dragid10) [inline _conferences/__main__.py]:
```suggestion
def _validate_issue(issue: gh_issues.Issue, date_to_check: str) -> bool:

```

File: tests/tests_conferences.py
Comment (dragid10) [inline tests/tests_conferences.py:88]:
```suggestion
    """tests the date validator against yesterday value (fail) and tomorrow value (pass)"""
```

File: _conferences/__main__.py
Comment (dragid10) [inline _conferences/__main__.py:69]: Something else that I think would make sense to check is if `end_date >= start_date`. Take this for example:

```
today_date = 10/05/2024
start_date = 10/15/2024
end_date   = 10/10/2024
```

This scenario will always clear the `validate_issue` check because the conference end date is after today's date. But it still shouldn't be considered a valid conference since the end date is before the start date. Let me know if I'm overthinking this one or not

File: _data/conferences.json
Comment (dragid10) [inline _data/conferences.json:10]: I'm curious, what will the format of the `speaking` field look like? Would it be a list of json objects?

**Review by dragid10 — CHANGES_REQUESTED** [review]
Awesome refactor! Just had some suggestions and a new test-case consideration

File: _conferences/__main__.py
Comment (kjaymiller) [reply _conferences/__main__.py:69]: I think maybe you're overthinking this as this validation is purely to filter out the entry from the generated json list.

These conference entries should still be validated by someone on the triage team and if there is an issue in entry, it is the triage members responsibility to detect that.

File: _data/conferences.json
Comment (kjaymiller) [reply _data/conferences.json:10]: Currently it is a  string of entries...

I was thinking of a GH-Issue feature that would that would convert a list to a python list.

with events.md being a .md file. It should render properly in the website.

**Review by dragid10 — APPROVED** [review]

**Comment by dragid10** [comment]
Well I'm good on these changes (though I don't have write access, so my approval means nothing)

**Comment by kjaymiller** [comment]
@dragid10 - updated the approval limit for the triage team

**Pull request merged**
